packages/@lwc/engine-core/src/framework/base-lightning-element.ts

```diff
@@ -459,6 +460,27 @@ LightningElement.prototype = {
         return getBoundingClientRect(elm);
     },
 
+    attachInternals(): ElementInternals {
+        const vm = getAssociatedVM(this);
+        const {
+            elm,
+            renderer: { attachInternals },
+        } = vm;
+
+        if (typeof ElementInternals === 'undefined') {
```

Comment:
One potential issue here: we are caching `HTMLElement.prototype.attachInternals` eagerly but evaluating `ElementInternals` dynamically. This could lead to mixed behavior if the polyfill loads after lwc does.

Let's be consistent and eagerly cache this. E.g. put this outside of the getter:

```js
const supportsElementInternals = typeof ElementInternals !== 'undefined'
```